src/Common/tests/System/PlatformDetection.cs

```diff
@@ -34,6 +34,58 @@ public static class PlatformDetection
 
         public static int WindowsVersion { get; } = GetWindowsVersion();
 
+        private static int s_isWinRT = -1;
```

Comment:
I am not sure if this a concern, you can just check if the Nullable has value and then if it didn't you initialize it. I am not expecting race condition will cause any harm in the initialization. the worst case it this will get initialized twice or so.